### PR TITLE
EA PG DatabaseMetadata - Part 1

### PR DESCRIPTION
**Pull request by oetherington**
This PR migrates the `DatabaseMetadata` collection to Postgres.

This should be merged at the same time as https://github.com/centre-for-effective-altruism/ForumCredentials/pull/12 in ForumCredentials.

Part 2 in #6646

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +46,7 @@
     "ea-start-prod-db": "./build.js -run --watch --settings ../ForumCredentials/settings-local-dev-prod-db.json --mongoUrlFile ../ForumCredentials/prod-db-conn.txt --postgresUrlFile ../ForumCredentials/prod-pg-conn.txt",
     "ea-start-testing-db": "./build.js -run --watch --settings ./settings-test.json --mongoUrl ${TESTING_DB_URL:=$(cat ../ForumCredentials/testing-db-conn.txt)} --postgresUrl ${PG_URL:=$(cat ../ForumCredentials/testing-pg-conn.txt)}",
     "init-cypress-db": "yarn ea-command 'Globals.dropAndCreatePg({templateId: \"cypress_template\", seed: true, dropExisting: true})'",
-    "init-cypress-db-ci": "./scripts/timeout.sh 120 yarn ea-command-ci 'Globals.dropAndCreatePg({templateId: \"cypress_template\", seed: true, dropExisting: true})'",
+    "init-cypress-db-ci": "yarn exec-accepted-schema && ./scripts/timeout.sh 120 yarn ea-command-ci 'Globals.dropAndCreatePg({templateId: \"cypress_template\", seed: true, dropExisting: true})'",
```
Comment (jpaddison3) [inline package.json:49]: Huh.

File: scripts/loadDbSettings.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  const db = pgPromiseLib({
+    connectionString,
+    max: 3,
+  });
+
+  await Promise.all([
+    insert(db, randomId(), "publicSettings", settings.databasePublicSettings),
+    insert(db, randomId(), "serverSettings", settings.databaseServerSettings),
+    deleteByName(db, "databaseId"),
```
Comment (jpaddison3) [inline scripts/loadDbSettings.ts:83]: This is only valid in the dev and staging cases. Please leave it in for prod.

File: scripts/loadDbSettings.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+type Settings = {
+  databasePublicSettings: Record<string, any>;
+  databaseServerSettings: Record<string, any>;
+}
+
+const isMode = (value: string): value is Mode =>
+  modes.includes(value as Mode);
+
+const capitalize = (s: string) => s[0].toUpperCase() + s.slice(1);
```
Comment (jpaddison3) [inline scripts/loadDbSettings.ts:21]: :yak:

File: scripts/loadDbSettings.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+const makeCredentialsPath = (fileName: string) =>
+  join("..", "..", "ForumCredentials", fileName);
+
+const readSettingsFile = (mode: Mode): Settings => {
+  if (mode === "xpost-dev") {
+    mode = "dev";
+  }
+  const path = makeCredentialsPath(`load${capitalize(mode)}DbSettings.js`);
+  const data = require(path);
```
Comment (jpaddison3) [inline scripts/loadDbSettings.ts:31]: You know, require is basically an eval, isn't it.

File: scripts/loadDbSettings.ts (hunk 2 above)
Comment (jpaddison3) [reply scripts/loadDbSettings.ts:83]: Future versions could let the file export the expected database id.

**Review by jpaddison3 — APPROVED** [review]

File: scripts/loadDbSettings.ts (hunk 2 above)
Comment (oetherington) [reply scripts/loadDbSettings.ts]: Good catch - should be fixed now to copy the behaviour from the old scripts

File: scripts/loadDbSettings.ts (hunk 4 above)
Comment (oetherington) [reply scripts/loadDbSettings.ts:31]: Yeah - _marginally_ less janky

**Pull request merged**